scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -214,7 +218,8 @@ class ScalametaParser(input: Input, dialect: Dialect) { parser =>
           if (curr.is[LeftParen]) ')' :: sepRegions
           else if (curr.is[LeftBracket]) ']' :: sepRegions
           else if (curr.is[LeftBrace]) '}' :: sepRegions
-          else if (curr.is[CaseIntro]) '\u21d2' :: sepRegions
+          //TODO: this must be handled better
```

Comment:
Added a test `test("case-match-interop".ignore) {` that shows a problem

When you parse `case ...` you ignore newlines until you find `=>` because syntax is `case ... => ...`

But with new enums it is no longer valid globally but only for 'match case'.
For 'enum case' newline actually terminates parsing.

For now made it to work for 'enum case' to test this and 'match case' is broken

Reply:
solved! thanks @tgodzik for snippets how to do it